lib/content-services/src/lib/tag/services/tag.service.spec.ts

```diff
@@ -70,5 +78,95 @@ describe('TagService', () => {
 
             service.addTag('fake-node-id', 'fake-tag');
         });
+
+        describe('createTags', () => {
+            it('should call createTags on tagsApi', () => {
+                spyOn(service.tagsApi, 'createTags').and.returnValue(Promise.resolve([]));
+                const tag1: TagBody = new TagBody();
+                tag1.tag = 'Some tag 1';
+                const tag2: TagBody = new TagBody();
+                tag2.tag = 'Some tag 2';
+                const tags: TagBody[] = [tag1, tag2];
+                service.createTags(tags);
+                expect(service.tagsApi.createTags).toHaveBeenCalledWith(tags);
+            });
+
+            it('should emit refresh when tags creation is success', fakeAsync(() => {
+                const tags: TagEntry[] = [{
+                    entry: {
+                        id: 'Some id 1',
+                        tag: 'Some tag 1'
+                    }
+                }];
+                spyOn(service.refresh, 'emit');
+                spyOn(service.tagsApi, 'createTags').and.returnValue(Promise.resolve(tags));
+                service.createTags([]);
+                tick();
+                expect(service.refresh.emit).toHaveBeenCalledWith(tags);
+            }));
+
+            it('should call error on logService when error occurs during tags creation', fakeAsync(() => {
+                const logService: LogService = TestBed.inject(LogService);
+                spyOn(logService, 'error');
+                const error: string = 'Some error';
+                spyOn(service.tagsApi, 'createTags').and.returnValue(Promise.reject(error));
+                service.createTags([]);
+                tick();
+                expect(logService.error).toHaveBeenCalledWith(error);
+            }));
+        });
+
+        describe('searchTags', () => {
+            let result: ResultSetPaging;
+
+            beforeEach(() => {
+                result = new ResultSetPaging();
+            });
+
+            it('should call search on searchApi with correct parameters', () => {
+                const searchSpy: jasmine.Spy<(queryBody: SearchRequest) => Promise<ResultSetPaging>> =
+                    spyOn(SearchApi.prototype, 'search').and.returnValue(Promise.resolve(result));
```

Comment:
we do not spy on prototypes, it's Angular, and there are means to inject and spy properly

Reply:
@DenysVuika I spied prototype because SearchAPI is not injectable so it is created in that way in service: 
private searchApi: SearchApi = new SearchApi(this.apiService.getInstance());

It is created directly by constructor. I can't spy instance of searchApi field because it is private. Can I leave it because of that as it is?